src/main/java/com/epam/test_generator/dao/interfaces/TestResultDAO.java

```diff
@@ -0,0 +1,28 @@
+package com.epam.test_generator.dao.interfaces;
+
+import com.epam.test_generator.entities.TestResult;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Date;
+import java.util.List;
+
+@Repository
+public interface TestResultDAO extends JpaRepository<TestResult, Long> {
+
+    /**
+     * @param projectId identifier of {@link com.epam.test_generator.entities.Project}
+     * @return Find all test results by projectId and make descending order by date.
+     */
+    List<TestResult> findAllByProjectIdOrderByDateDesc(long projectId);
+
+    /**
+     *
+     * @param projectId identifier of {@link com.epam.test_generator.entities.Project}
+     * @param after Find all test results which are after than this.
+     * @param before Find all test results which are before than this.
+     * @return Find all test results by projectId and which dates are lean between after and before.
+     */
+    List<TestResult> findAllByProjectIdAndDateAfterAndDateBefore(long projectId, Date after, Date before);
```

Comment:
https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation

Use `Between` instead

Reply:
`findAllByProjectIdAndDateBetween(long projectId, Date start, Date end)`  and `findAllByProjectIdAndDateIsBetween(long projectId, Date start, Date end)` haven't worked for some reason. Is it allowable to leave it this way?

Reply:
Yes, let it be